### PR TITLE
Bids 1537/prater exporter eden relay not working

### PR DESCRIPTION
**Pull request by Eisei24**
We don't use `NOW() AT TIME ZONE 'utc'` anywhere else but it is important here so that we can use `time.Since(...)`.
Since every entry in the db is `timestamp without time zone` if you just save it with `NOW()` it will be the local time and then for `time.Since(...)` it will assume it to be UTC.

It was quite complicated to test this, I create relay entries in my own local prater db (not the normal prater one).

### CODE REVIEW

File: exporter/relays.go [hunk 1]
```diff
@@ -31,32 +32,75 @@ func mevBoostRelaysExporter() {
...
 		return
 	}
+
+	mux.Lock()
+	_, err = db.WriterDb.Exec(`
+			UPDATE relays SET
+				export_failure_count = $1,
+				last_export_try_ts = (NOW() AT TIME ZONE 'utc'),
+				last_export_success_ts = (NOW() AT TIME ZONE 'utc')
+			WHERE tag_id = $2 AND endpoint = $3`, 0, r.ID, r.Endpoint)
```
Comment (D13ce) [inline exporter/relays.go:98]: **Minor:**

```suggestion
	_, err = db.WriterDb.Exec(`
			UPDATE relays SET
				export_failure_count = 0,
				last_export_try_ts = (NOW() AT TIME ZONE 'utc'),
				last_export_success_ts = (NOW() AT TIME ZONE 'utc')
			WHERE tag_id = $1 AND endpoint = $2`, r.ID, r.Endpoint)
```
:)

**Review by D13ce — COMMENTED** [review]
utACK

I have left some minor suggestions though. Please respond before I approve.

File: exporter/relays.go [hunk 2]
```diff
@@ -31,32 +32,75 @@ func mevBoostRelaysExporter() {
...
-func singleRelayExport(r types.Relay, wg *sync.WaitGroup) {
+func singleRelayExport(r types.Relay, wg *sync.WaitGroup, mux *sync.Mutex) {
 	defer wg.Done()
+
 	err := exportRelayBlocks(r)
 	if err != nil {
-		r.Logger.Errorf("failed to export blocks for relay: %v", err)
+		errMsg := fmt.Errorf("failed to export blocks for relay: %v", err)
+		if shouldLogExportAsError(r) {
+			r.Logger.Error(errMsg)
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply exporter/relays.go:64]: We log something within the relay itself, isn't that a completely different kind of logging?
Should this be changed?

File: exporter/relays.go (hunk 1 above)
Comment (Eisei24) [reply exporter/relays.go:98]: Sure, I now like it a bit more.

File: exporter/relays.go [hunk 3]
```diff
@@ -225,3 +269,30 @@ func retrieveAndInsertPayloadsFromRelay(r types.Relay, low_bound uint64, high_bo
...
+	maxWaitTimeForRelayExport := utils.Day
+
+	waitTime := time.Duration(math.Exp2(float64(r.ExportFailureCount))) * time.Minute
+	isMaxWaitTime := false
+	if waitTime >= maxWaitTimeForRelayExport {
+		waitTime = maxWaitTimeForRelayExport
+		isMaxWaitTime = true
+	}
+	return waitTime, isMaxWaitTime
+}
```
[the remark this answers is not in the document]
Comment (D13ce) [reply exporter/relays.go]: As discussed, you are referring to the empty return statement. Even if the function would be longer, you could still use named return values and simply write
```
return waitTime, isMaxWaitTime
```
at the end.

**Review by D13ce — APPROVED** [review]
utACK

**Pull request merged**